### DISCUSSION TITLE
Blade f7 hd motor issue

### DISCUSSION THREAD

**Discussion by xpertcrush** [iNavFlight/inav#7849] [Q&A]
## Current Behavior
In motors tab when I use it to test motors they just spin like 5 % throttle. I tried various settings, dshot multishot all have same result. Except motor 2 which responds correctly to the throttle.

## Expected behavior
It should be linear response to the throttle.

## Suggested solution(s)
These are my current settings and i don't see a conflict on dma side.

![P_20220210_175503.jpg](https://user-images.githubusercontent.com/67045883/153408455-c2e410af-3876-4828-9a24-144467ed7601.jpg)
Even I manually assigned dma to the pins
![P_20220210_175643.jpg](https://user-images.githubusercontent.com/67045883/153408477-c7dac848-ea6c-46cd-b2f6-e1d07f194f87.jpg)

## Additional context

---

- FC Board name and vendor:
- INAV version string:
4.1
Compiled on windows 10

![P_20220210_175819.jpg](https://user-images.githubusercontent.com/67045883/153408682-bd281665-dd0c-4a56-b40b-15440e805272.jpg)

**Comment by Odin263** [comment]
Are you sure you are using the right target? I can't find anything in iNav 4.1 that matches your images. If you really have the RUSH BLADE F7 board, you might want to try the following target:
[inav_4.1.0_RUSH_BLADE_F7.zip](https://github.com/iNavFlight/inav/files/8043524/inav_4.1.0_RUSH_BLADE_F7.zip)

By the way, in Betaflight only MS5611 is entered as baro in target.h. BMP280, MS5611, DPS310 are available in iNav. Maybe the MS5611 is built into your board?

**Reply by KelkieFPV** [reply]
thankyou so much mate i been searching for this for a long time was about to give up. please can you get this put into main INAV list
thanks again mate.

**Reply by KelkieFPV** [reply]
no baro tho.

**Comment by xpertcrush** [comment]
> Are you sure you are using the right target? I can't find anything in iNav 4.1 that matches your images. If you really have the RUSH BLADE F7 board, you might want to try the following target:
> [inav_4.1.0_RUSH_BLADE_F7.zip](https://github.com/iNavFlight/inav/files/8043524/inav_4.1.0_RUSH_BLADE_F7.zip)
> 
> By the way, in Betaflight only MS5611 is entered as baro in target.h. BMP280, MS5611, DPS310 are available in iNav. Maybe the MS5611 is built into your board?
> 
> 

Yes mine is blade f7 hd version it doesn't have osd.


I did those changes, it wasn't there in inav.


Motors don't work as intended, does it have anything to do with blheli s? On betaflight baro is detected as qmp6988. On betaflight everything just works, but not on inav. I am out of options, my friend has blade f7, I compiled a version for him and it's working fine for him, but unfortunately mine isn't working correctly.
